src/PlayerManager.cpp

```diff
@@ -723,6 +724,21 @@ void PlayerManager::changePlayerGuide(CNSocket *sock, CNPacketData *data) {
     sock->sendPacket((void*)&resp, P_FE2CL_REP_PC_CHANGE_MENTOR_SUCC, sizeof(sP_FE2CL_REP_PC_CHANGE_MENTOR_SUCC));
 }
 
+void PlayerManager::setFirstUseFlags(CNSocket* sock, CNPacketData* data) {
+    if (data->size != sizeof(sP_CL2FE_REQ_PC_FIRST_USE_FLAG_SET))
+        return;         //ignore malformed packet
+    sP_CL2FE_REQ_PC_FIRST_USE_FLAG_SET* flag = (sP_CL2FE_REQ_PC_FIRST_USE_FLAG_SET*)data->buf;
+    Player* plr = getPlayer(sock);
+    if (flag->iFlagCode <= 64) {
+        int newflag = 1ULL << (flag->iFlagCode - 1);
```

Comment:
```suggestion
        uint32_t newflag = 1UL << (flag->iFlagCode - 1);
```
These bitfields are 32-bit, so ULL isn't necessary and would only get truncated upon assignment.
On the other had, `uint32_t` must be used to make sure the variable is 32-bit and *unsigned*.

Reply:
As for the int, I just imitated the client code and it used a `long` instead of int so okay I will change it like you said.
As for uint32_t, I think to keep it consistent with the right hand side, it should be int64_t